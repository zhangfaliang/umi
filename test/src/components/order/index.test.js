import { OrderCard, OrderLabelCard, OrderFilter } from '.';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

const Panel = OrderFilter.Panel;
const Card = OrderLabelCard.Card;
const Item = OrderCard.Item;

describe('order组件', async () => {
  it('OrderLabelCard', () => {
    const wrapper = render(
      <OrderLabelCard label={'Royal Pari FC 2.8'}>
        <Card>
          <Item extra={122} size="small">
            order:222
          </Item>
          <Item extra={122} size="small">
            order:222
          </Item>
          <Item size="small">order:222</Item>
        </Card>
        <Card>
          <Item extra={122} size="small">
            order:222
          </Item>
        </Card>
      </OrderLabelCard>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('OrderCard', () => {
    const wrapper = render(
      <OrderCard label="Order information">
        <Item extra={'match'}>Royal Pari FC VS Nacional Potosi</Item>
        <Item extra={'Match Result'}>Market</Item>
      </OrderCard>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('OrderFilterRender', () => {
    const wrapper = render(
      <OrderFilter defaultActiveKey={['0', '2']}>
        <Panel body={[{ name: 'Last 24 Hrs' }, { name: 'Last 48 Hrs' }, { name: 'Last 6 Mths' }]} />
        <Panel body={[{ name: 'All' }, { name: 'Settled' }, { name: 'Unsettled' }]} />
      </OrderFilter>
    );
    expect(toJson(wrapper));
  });

  it('click', () => {
    const wrapper = mount(
      <OrderFilter defaultActiveKey={['0', '2']}>
        <Panel body={[{ name: 'Last 24 Hrs' }, { name: 'Last 48 Hrs' }, { name: 'Last 6 Mths' }]} />
        <Panel body={[{ name: 'All' }, { name: 'Settled' }, { name: 'Unsettled' }]} />
      </OrderFilter>
    );
    wrapper
      .find('#orderFilterTap')
      .at(0)
      .find('span')
      .simulate('click');
    wrapper
      .find('#orderFilterTap')
      .at(1)
      .find('span')
      .simulate('click');
    expect(
      wrapper
        .find('#orderFilterTap')
        .at(0)
        .find('span')
        .hasClass('hight')
    ).toEqual(true);
  });

  it('onChange', () => {
    const wrapper = mount(
      <OrderFilter defaultActiveKey={['0', '2']}>
        <Panel body={[{ name: 'Last 24 Hrs' }, { name: 'Last 48 Hrs' }, { name: 'Last 6 Mths' }]} />
        <Panel body={[{ name: 'All' }, { name: 'Settled' }, { name: 'Unsettled' }]} />
      </OrderFilter>
    );
    wrapper
      .find('#orderFilterTap')
      .at(0)
      .find('span')
      .simulate('click');
    wrapper
      .find('OrderFilter')
      .at(0)
      .find('.body')
      .find('.orderFilterActiveTap')
      .at(0)
      .simulate('click');
    expect(
      wrapper
        .find('OrderFilter')
        .at(0)
        .find('.body')
        .find('.orderFilterActiveTap')
        .at(0)
        .text()
    ).toEqual('Last 24 Hrs');
  });
});
