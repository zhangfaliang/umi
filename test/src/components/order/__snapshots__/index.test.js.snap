// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`order组件 OrderCard 1`] = `
<div
  class="wrap"
>
  <div
    class="label"
  >
    Order information
  </div>
  <div
    class="list"
  >
    <div
      class="item large"
    >
      <div
        class="item_L"
      >
        match
      </div>
      <div
        class="item_R"
      >
        Royal Pari FC VS Nacional Potosi
      </div>
    </div>
    <div
      class="item large"
    >
      <div
        class="item_L"
      >
        Match Result
      </div>
      <div
        class="item_R"
      >
        Market
      </div>
    </div>
  </div>
</div>
`;

exports[`order组件 OrderFilterRender 1`] = `
<div
  class="wrap"
>
  <div
    class="nav"
  >
    <div
      class="item"
    >
      <div
        class="header"
      >
        <span
          class=""
          id="orderFilterTap"
        >
          Last 24 Hrs
          <img
            alt=""
            class=""
            src=""
          />
        </span>
      </div>
    </div>
    <div
      class="item"
    >
      <div
        class="header"
      >
        <span
          class=""
          id="orderFilterTap"
        >
          Unsettled
          <img
            alt=""
            class=""
            src=""
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`order组件 OrderLabelCard 1`] = `
<div
  class="wrap"
>
  <div
    class="label"
  >
    Royal Pari FC 2.8
  </div>
  <div
    class="body"
  >
    <div>
      <div
        class="item small"
      >
        <div
          class="item_L"
        >
          122
        </div>
        <div
          class="item_R"
        >
          order:222
        </div>
      </div>
    </div>
    <div>
      <div
        class="item small"
      >
        <div
          class="item_L"
        >
          122
        </div>
        <div
          class="item_R"
        >
          order:222
        </div>
      </div>
    </div>
    <div>
      <div
        class="item small"
      >
        <div
          class="item_R"
        >
          order:222
        </div>
      </div>
    </div>
  </div>
  <div
    class="body"
  >
    <div>
      <div
        class="item small"
      >
        <div
          class="item_L"
        >
          122
        </div>
        <div
          class="item_R"
        >
          order:222
        </div>
      </div>
    </div>
  </div>
</div>
`;
